### PR TITLE
Add support for Arista LB2 QDMA shell

### PR DESCRIPTION
**Pull request by kennylau-arista**
- Add Arista PCIe IDs
- Support for vendor-dependent firmware paths
- Reduce dmatest memory usage for Arista devices when running `xbutil validate`
- Different flash management in `xbmgmt` for Arista devices

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

**Comment by maxzhen** [comment]
@kennylau-arista , thanks for filing this pull request to add more boards support to XRT. We will review the change and get this merged ASAP.

File: src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp [hunk 1]
```diff
@@ -1932,9 +1962,27 @@ static int bitstreamGuardAddress(pcidev::pci_device *dev, uint32_t& addr)
...
+    // Vendor-dependent behaviour
+    mDev->sysfs_get<uint16_t>("", "vendor", errmsg, vendor, -1);
+    if (!errmsg.empty()) {
+        std::cout << errmsg << std::endl;
+        return -EINVAL;
+    }
+
+    switch (vendor) {
+        case ARISTA_ID:
+            return fullErase() ? 0 : -EINVAL;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp:1978]: The concept of revertToMFG is to have a valid shell locked down on the flash always. This shell will take effect when there is no other valid shell on flash. When you erase the whole flash, do you still have a valid shell?

Comment (kennylau-arista) [reply src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp:1978]: In our case we do not have a golden image stored in the flash, the default state is an empty flash.

Comment (maxzhen) [reply src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp:1978]: > 
> 
> In our case we do not have a golden image stored in the flash, the default state is an empty flash.

OK, I see. Thanks.

**Comment by maxzhen** [comment]
retest this please

**Comment by maxzhen** [comment]
@kennylau-arista , please remove the '/' from your branch name. that might be causing build failures...

**Comment by kennylau-arista** [comment]
> @kennylau-arista , please remove the '/' from your branch name. that might be causing build failures...

Sure, the new PR is at #3875

**Pull request closed without merging**